codegen/src/main/scala/akka/grpc/gen/Main.scala

```diff
@@ -46,18 +50,34 @@ object Main extends App {
 
   val out = {
     val codeGenerator =
-      if (languageScala) {
-        // Scala
-        if (generateClient && generateServer) ScalaBothCodeGenerator
-        else if (generateClient) ScalaClientCodeGenerator
-        else if (generateServer) ScalaServerCodeGenerator
-        else throw new IllegalArgumentException("At least one of generateClient or generateServer must be enabled")
+      if (!generatePlay) {
+        if (languageScala) {
+          // Scala
+          if (generateClient && generateServer) ScalaBothCodeGenerator
+          else if (generateClient) ScalaClientCodeGenerator
+          else if (generateServer) ScalaServerCodeGenerator
+          else throw new IllegalArgumentException("At least one of generateClient or generateServer must be enabled")
+        } else {
+          // Java
+          if (generateClient && generateServer) JavaBothCodeGenerator
+          else if (generateClient) JavaClientCodeGenerator
+          else if (generateServer) JavaServerCodeGenerator
+          else throw new IllegalArgumentException("At least one of generateClient or generateServer must be enabled")
+        }
       } else {
-        // Java
-        if (generateClient && generateServer) JavaBothCodeGenerator
-        else if (generateClient) JavaClientCodeGenerator
-        else if (generateServer) JavaServerCodeGenerator
-        else throw new IllegalArgumentException("At least one of generateClient or generateServer must be enabled")
+        if (languageScala) {
+          // Scala
+          if (generateClient && generateServer) PlayScalaClientCodeGenerator
+          else if (generateClient) PlayScalaClientCodeGenerator
+          else if (generateServer) PlayScalaServerCodeGenerator
+          else throw new IllegalArgumentException("At least one of generateClient or generateServer must be enabled")
+        } else {
+          // Java
+          if (generateClient && generateServer) PlayJavaBothCodeGenerator
+          else if (generateClient) PlayJavaClientCodeGenerator
```

Comment:
If I'm reading this right, we're running *either* `JavaClientCodeGenerator` *or* `PlayJavaClientCodeGenerator` here. For sbt, I think we're running `JavaClientCodeGenerator` and *optionally also* `PlayJavaClientCodeGenerator`: the code generated by `PlayJavaClientCodeGenerator` depends on the code generated by `JavaClientCodeGenerator`, so running only `PlayJavaClientCodeGenerator` might not be sufficient.

I think we have 2 ways to approach this: 
* invoking this `Main` twice from the Gradle plugin, once with `generate_play` and once without
* invoking this `Main` once from the Gradle plugin, and generate both the 'plain' client and the play-specific code at once.

I think the second approach is ultimately the most appealing one, as it allows eventually splitting off the play-specific portions.

For MVP, however, the first approach is likely the most easy/practical. It might even be as simple as introducing a `object CombinedPlayJavaClientCodeGenerator extends PlayJavaClientCodeGenerator with JavaClientCodeGenerator`. We should make sure this is a private generator though, to avoid people accidentally using it from sbt.

Reply:
Let's go with the good enough one, please ticketify the other style

Reply:
Not entirely sure how to do this in the plugin to be honest... Let's chat and I can PR and merge it later on on top of this. @raboof